Allow multiple files with the same base names to pass header inclusion test

crogre opened the pull request:
This closes #102.

mwoehlke-kitware committed:
Allow space before C++11 attributes

Modify rule checking for spaces before `[`s to ignore C++11 attributes,
which usually should have spaces. This fixes an incorrect warning on
code like:

  #define ABIEXPORT [[gnu::visibility("default")]]

rjmunro committed:
Add sed and gsed output formats to auto-fix code

The sed format outputs sed commands that should fix some of the errors.
Note that this requires gnu sed. If that is installed as gsed on your
system (common e.g. on macOS with homebrew) you can use the gsed output
format. Sed commands are written to stdout, not stderr, so you should be
able to pipe output straight to a shell to run the fixes.

Commit:
Fixed --root option under Windows

backes committed:
Extend class decl regexp to match V8's decorators

This also allows digits in class decorators, to also match macros like
V8_EXPORT_PRIVATE.

groegeorg committed:
Fixed search for repository root directory (#78)

In the situation, where one git/svn/hg repository is located inside the
directory tree of another, the search incorrectly selects the outer one
instead of the inner one.

tkruse committed:
Fix quotes and raw strings

geoffviola committed:
differentiate namespace headers

tkruse committed:
add cli test for boost header with anonymous namespace

tkruse commented:
This pull request seems wrong. It wants to merge from cpplint:master into cogre:master, not the other way round. The correct way to create a pull request would be to go to https://github.com/cpplint/cpplint, click "New pull request", and then select the cogre branch .

crogre commented:
Sorry, I got things mixed up. New, hopefully correct PR is up!

Greger

> 
> 31 aug. 2019 kl. 11:36 skrev Thibault Kruse <notifications@github.com>:
> 
> ﻿
> This pull request seems wrong. It wants to merge from cpplint:master into cogre:master, not the other way round. The correct way to create a pull request would be to go to https://github.com/cpplint/cpplint, click "New pull request", and then select the cogre branch .
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Commit:
add list of c headers

tkruse committed:
NOBUG: fix inernal docs

tkruse committed:
Merge branch 'sample-licenses' into develop

Shachlan committed:
Fix #135: allow 'if constexpr' in readability/braces.

vapier committed:
Project import generated by Copybara.

PiperOrigin-RevId: 306352475

IsaacG committed:
Merge pull request #544 from google/change_HEAD

Project import generated by Copybara.

Commit:
Add a canonical link to update the URL for the Shell guide

IsaacG committed:
Merge pull request #546 from google/shell_canonical

Add a canonical link to update the URL for the Shell guide

tkruse committed:
Try fix travis failure

tkruse committed:
Documentation: Add reference to discussion

tkruse committed:
Fix travis by freezing pyparsing version

tkruse committed:
freeze versions for python 2.7, fix setup.py for commments breaking pip

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 311271037

IsaacG committed:
Merge pull request #549 from google/change_HEAD

Project import generated by Copybara.

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 312117514

cottsay committed:
Close files after reading contents

Addresses ResourceWarning messages in console output when running with
deubg-enabled Python.

zetafunction committed:
Update C++ styleguide

Major changes:
- discourage const references when the input parameter must outlive the
  call
- drop ban on mutable references
- allow rvalue-qualified methods in certain situations
- allow C-style cast when casting to void
- disallow postfix increment/decrement operators unless postfix
  semantics are required
- allow designated initializers in C++20-compliant form
- disallow macro-style naming for enumerator constant for new code
- disallow spaces inside parentheses for conditionals

The change itself is largely automated, with a few manual adjustments:
- manually restore the TOC placeholder
- removed trailing whitespace

tkruse committed:
NOBUG: Fix lint error in setup.py

tkruse committed:
close headerfile after reading

tituswinters committed:
Merge pull request #553 from zetafunction/cpp-styleguide-update

Update C++ styleguide

zetafunction committed:
[C++ styleguide] Restore table of contents

This was mistakenly dropped when updating the styleguide from the
internal C++ styleguide.

sfreilich committed:
Merge pull request #375 from ebzzry/gh-pages

Fix typos in the lisp guide

sfreilich committed:
Merge pull request #266 from svetlyak40wt/patch-1

Fixed "full hairy example" of formatting control string in Common Lisp style guide

sfreilich committed:
Merge pull request #334 from ebzzry/gh-pages

Fix typo

tituswinters committed:
Merge pull request #555 from zetafunction/restore-toc

[C++ styleguide] Restore table of contents

tkruse committed:
Releasing 1.4.6

tkruse committed:
NOBUG: minor documentation fix

tkruse committed:
Release 1.5.0

tkruse committed:
NOBUG: tox with pythonh 3.7 and 3.8, doc changes

tkruse committed:
Unit test for decorator-tolerant parsing

tkruse committed:
Avoid false positive warnings for QT signals and slots extended syntax.

This reuses the same regex snippet as in Update()

Note that QT also allows to use macros instead of keywords: Q_SIGNAL, Q_SIGNALS, Q_SLOT, Q_SLOTS, but apparently this is not used not frequently enough to include in cpplint.

tkruse committed:
Fix #146: Additional Linux glibc headers not recognized as C System headers

jbcoe committed:
Add Google C# style guide

tkruse committed:
2nd attempt for #43: Hide new include order behind new cli flag --includeorder=standardcfirst

tituswinters committed:
Merge pull request #559 from jbcoe/gh-pages

Add Google C# style guide

jbcoe committed:
Remove markdown not recognised by GitHub

Remove `{.good}` and `{.bad}` and replace with comments.

jbcoe committed:
Remove more markdown that GitHub cannot handle

tkruse committed:
Release 1.5.1

jameslamb committed:
Add language about Unix directory aliases (fixes #139)

jameslamb committed:
try to fix tests

jameslamb committed:
remove specific mention of Unix

jameslamb committed:
[docs] minor fixes to CONTRIBUTING.rst

jameslamb committed:
fix tests

nealrichardson committed:
Remove blacklist/whitelist from cpplint.py

gpshead committed:
Remove blacklist/whitelist from cpplint.py (#562)

Remove blacklist/whitelist from cpplint.py

Merges pull request #562 from @nealrichardson.

tkruse committed:
Merge remote-tracking branch 'google/gh-pages'

tkruse committed:
small improvements to CONTRIBUTING

tkruse committed:
NOBUG: Improve CLITest output

tkruse committed:
FIX: Sort collected files before checking to have repeatable output

tkruse committed:
Release 1.5.2

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 318489724

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 318884292

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 320107176

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 320248352

jameslamb committed:
[ci] use portable temp directory approach in tests (fixes #154)

Jasper-Ben committed:
allow usage of sed

tkruse committed:
NOBUG: Add code doc

tkruse committed:
Fix #156: Suppress Info output in sed/gsed format to allow easy piping

tkruse committed:
add CLI test for sed output

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 321580967

tkruse committed:
Release 1.5.3

sfreilich committed:
Add CODEOWNERS file

Since this repository contains style guides managed by different owners/teams, it might be convenient to have a way to allow maintainers to get notified about PRs for specific files. GitHub's code owners feature provides a way to do that:
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

Adding myself to this for the Lisp Style Guide.

sfreilich committed:
Merge pull request #569 from google/sfreilich-codeowners

Add CODEOWNERS file

jbcoe committed:
Add jbcoe as CODEOWNER for C# style guide

gpshead committed:
own Python things.

allight committed:
Update language to comply with Android's inclusive language guidance

See https://source.android.com/setup/contribute/respectful-code for
reference

cpplint.py is copied into AOSP from this github version so update it
here first.

Bug: 161896447
Test: None (comments only)

gpshead committed:
Merge pull request #570 from allight/inclusive

Update language to comply with Android's inclusive language guidance (remove "dummy").

tonyruscoe committed:
Add CODEOWNER for HTML/CSS style guide

ptomato committed:
Unit test for spacing before square brackets

Testing in particular that C++11 attributes are not flagged with this
warning.

tkruse committed:
Merge remote-tracking branch 'google/gh-pages' into develop

tkruse committed:
changelog

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 325285627

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 325456551

tkruse committed:
upgrade version

tkruse committed:
doc change

tkruse committed:
changelog

dimo414 committed:
Add owners for the Java and Shell guides, and alphabetize the list.

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 332244028

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 332885552

zetafunction committed:
Update C++ styleguide

- add section for inclusive language.
- move include-what-you-use to a new section.
- clarify usage of absl::optional for by-value vs by-reference params.
- document usage of `` as a code-indexing hint
- note that enclosing identifiers in `` can help code indexers.

tituswinters committed:
Merge pull request #577 from zetafunction/how-does-this-work-again

Update C++ styleguide

FaresSalem committed:
Update style guideline URL

zetafunction committed:
Update Google C++ styleguide

- Guidance for return values (prefer return by value, return by
  reference, then return by pointer in that order)
- Mandate curly braces around the body of a control statement,
  providing one exception for historical reasons (`if` with no `else`,
  and `if` + body fits completely in a single line or the `if` fits in
  one line and the body fits in another separate line).

tituswinters committed:
Merge pull request #584 from zetafunction/yet-again-how-does-this-work

Update C++ styleguide

Migelo committed:
add elif as an exception for CheckSpacingForFunctionCall()

nvuillam committed:
Import fixes for CI + update doc link

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 347693939

alex-okrushko committed:
Add TypeScript style guide

Migelo committed:
add the accompanying unit test

Migelo committed:
fix CI for py35, py27 and pypy

tkruse committed:
NOBUG: run tox using py38

alex-okrushko committed:
remove the go-link

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 353740685

zetafunction committed:
Update C++ style guide.

Minor formatting and wording fixes.

tituswinters committed:
Merge pull request #620 from zetafunction/update-c++

Update C++ style guide.

OlehKSS committed:
Pre-commit hook was added.

OlehKSS committed:
Put correct name.

PhilLab committed:
Fixed wrong CLI help text: Each filter needs + or -

AddFilters() raises the error:
  'Every filter in --filters must start with + or -'

zetafunction committed:
Update C++ style guide.

Mostly minor wording updates (e.g. preferring int64_t to int64). The
most significant change is clarifying the section on consistency to
clarify that consistency, in and of itself, should not be used as the
sole argument for blocking adoption of style changes.

tituswinters committed:
Merge pull request #630 from zetafunction/gh-pages

Update C++ style guide.

Sidelobe committed:
Added 'size_t' to typecasts detected by CheckCStyleCast

elharo committed:
docs: discourage external contributors

@tituswinters

elharo committed:
expand on treatment of external contributions

elharo committed:
Update README.md

tituswinters committed:
Merge pull request #633 from elharo/patch-1

docs: discourage external contributors

elharo committed:
Merge pull request #1 from google/gh-pages

merge head

Commit:
add pull request template discouraging contributions

Commit:
allow for typo corrections

tkruse committed:
fix py27 on travis by freezing pytest-runner

Sidelobe committed:
Added unit test for size_t C-Style cast check

tituswinters committed:
Merge pull request #634 from elharo/gh-pages

add pull request template discouraging contributions

Commit:
add code of conduct

Commit:
add issue template

tituswinters committed:
Merge pull request #636 from elharo/issue

add issue template discouraging contributions

Commit:
use the 2.0 version of the Contributor Covenant, with modification

tituswinters committed:
Merge pull request #635 from elharo/code

add code of conduct

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 372441803

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 372845522

nvuillam committed:
Sort extension list in help (today the order is random)

tkruse committed:
NOBUG: Pylint, ignore minor style warning

tkruse committed:
Adapt integration tests for #172

tkruse committed:
Releasing 1.5.5: Bump version, Update changelog

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 384778037

jruderman committed:
cppguide: remove space before comma

tituswinters committed:
Merge pull request #643 from jruderman/patch-1

cppguide: remove space before comma

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 385825978

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 396282811

zetafunction committed:
Update the C++ style guide.

- Recommend the use of absl::implicit_cast for safe upcasting.
- Remove std::hash guidance
- Remove end-of-line comment guidance
- Remove "consistency to local style" guidance
- Miscellaneous wording updates

tituswinters committed:
Merge pull request #650 from zetafunction/new-thingie

Update the C++ style guide.

wjwwood committed:
ignore NOLINT comments with categories that come from clang-tidy

Signed-off-by: William Woodall <william@osrfoundation.org>

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 414768544

gpshead committed:
Merge pull request #660 from google/python_styleguide

Project import generated by Copybara.

tonyruscoe committed:
Remove prettyprint from HTML/CSS Style Guide

This is because the internal style guide does not use prettyprint.

tonyruscoe committed:
Merge pull request #666 from tonyruscoe/gh-pages

Remove prettyprint from HTML/CSS Style Guide

tonyruscoe committed:
Replace HTML entities in the HTML/CSS Style Guide

The internal script that converts our style guides from Markdown to HTML now does this.

tonyruscoe committed:
Update format for numbered sections and whitespace

tonyruscoe committed:
Merge pull request #667 from tonyruscoe/gh-pages

Update format for numbered sections and whitespace

tonyruscoe committed:
Explain the 4-space HTML line-wrapping guidance

This clarification was added to the internal style guide a few months ago.

tonyruscoe committed:
Prefer leading zeros in CSS values

This was reversed in the internal style guide a while ago to align the guidance with our autoformatting tooling configuration.

Any optimisations to reduce file size should be handled by compilers and minifiers.

tonyruscoe committed:
Move the ID and Class Name sections to be together

This change was made to the internal style guide a while ago to make it easier to find related sections also containing IDs and class name guidance.

tonyruscoe committed:
Update ID and class selector/attribute guidance

We now recommend avoiding unnecessary `id` attributes and prefer using class selectors for styling.

Where `id` attributes are strictly required, we recommend always including a hyphen in the value.

tonyruscoe committed:
Add recommendation to avoid using !important

This recommendation was adopted internally to align with our infrastructure conformance checks and best practices.

While !important is a legitimate feature of CSS, its use increased complexity and maintenance costs when styles were reused and shared across multiple projects.

cushon committed:
Update Java style guide

cushon committed:
Move a </p> around

dimo414 committed:
Merge pull request #671 from cushon/gh-pages

Update Java style guide

geoffviola committed:
handle other header files for build include subdir

build/include_subdir should handle arbitrarily header file extensions.

Tests
include "test.hh" // now emits warning with defaults
include "test.h"  // same warning with defaults
include "test.aa" // can emit warning if .aa set as header

rex4539 committed:
Fix typos

sfreilich committed:
Merge pull request #673 from rex4539/typos

Fix typos

tkruse committed:
Add github action because travis-ci shut down

tkruse committed:
remove travis integration because travis-ci shut down

tkruse committed:
remove travis references

tkruse committed:
remove py 3.10 because of failure

tkruse committed:
fix lint failures with new pylint versions

tkruse committed:
Fix flake8 setuptools support removed, adapt tests to flake8 rules

tkruse committed:
Fix sample regression tests

tkruse committed:
changelog for version 1.6.0

tkruse committed:
try fixing PR action

armandas committed:
Fix post increment/decrement operator causing a false positive. (#195)

* Fix post increment/decrement operator causing a false positive.
Co-authored-by: Armandas Jarušauskas <armandas.jarusauskas@zmp.co.jp>

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 440926376

gpshead committed:
Merge pull request #683 from google/python_styleguide

Project import generated by Copybara.  Mostly modernizing to a 3.9+ Pythonisms. Some clarification on TypeVars.

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 442743475

zetafunction committed:
Update C++ style guide.

- Reword guidance for definitions of inline functions and templates.
- Capitalize the first letter of various comments.
- Update command-line flag reference to use absl flags.
- Improve braced-initialization example for 64-bit constants.
- Delete C++03 guidance for NULL.
- Clarify wording around wrapping of string literals.
- Use C++17 [[fallthrough]] instead of ABSL_FALLTHROUGH.

tonyruscoe committed:
Update HTML Line-Wrapping examples

Replaces the examples in the HTML Line-Wrapping recommendation to include child elements.

tituswinters committed:
Merge pull request #688 from zetafunction/update-styleguide

Update C++ style guide.

tonyruscoe committed:
Make the Declaration Order recommendation optional

This recommendation was relaxed in the internal style guide a few months ago. The suggestion is still to continue using alphabetical order unless teams have access to the appropriate tooling which can support consistently automating and enforcing different sorting rules which may be more logical to read.

jacobperron committed:
Files with '.hh' extension are not system headers

Support for checking .hh headers by default was added in
https://github.com/cpplint/cpplint/commit/3d8f6f876dd6e3918e5641483298dbc82e65f358.

However, without this change .hh files being included are considered to be C system headers.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 450985691

gpshead committed:
Merge pull request #692 from google/python_styleguide

Project import generated by Copybara.

tkruse committed:
Files with '.hh' extension are not system headers. Add cli test

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 453553841

gpshead committed:
Merge pull request #697 from google/python_styleguide

Project import generated by Copybara.

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 453566611

zetafunction committed:
Update C++ style guide.

- Include friend types in class declaration order guidance.
- Include previously omitted text (due to mismatched tags) about
  preferring int16_t over short, et cetera.
- Function declarations:
  - Updated guidance for what comments should cover.
  - Add a C++ attribute example.

tituswinters committed:
Merge pull request #699 from zetafunction/moo

Update C++ style guide.

sangmo-kang committed:
Fix failure when opening CPPLINT.cfg in python2

encoding is an invalid argument for open in python2.

gpshead committed:
Merge pull request #580 from FaresSalem/gh-pages

Update style guideline URL

tkruse committed:
Add integration test for using CPPLINT.cfg file

tkruse committed:
Add regression test for _OTHER_NOLINT_CATEGORY_PREFIXES

tkruse committed:
Release 1.6.1, provide changelog

tkruse committed:
fix pylint warnings

tkruse committed:
fix tox.ini

tkruse committed:
more flags

tkruse committed:
Merge remote-tracking branch 'google/gh-pages' into develop

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 480954960

phoe committed:
Fix PRINT-UNPRINTABLE-OBJECT

There is no Common Lisp operator named PRINT-UNPRINTABLE-OBJECT. Given the context, it is certain that PRINT-UNREADABLE-OBJECT (www.lispworks.com/documentation/lw51/CLHS/Body/m_pr_unr.htm) was meant instead.

sfreilich committed:
Merge pull request #719 from phoe-trash/gh-pages

Fix PRINT-UNPRINTABLE-OBJECT

Cabbageboy committed:
update ts styleguide with latest google version

sfreilich committed:
Merge pull request #727 from Cabbageboy/gh-pages

Update ts styleguide with latest google version

gaal committed:
Add Google Go Style Guide.

PiperOrigin-RevId: 489165518

tituswinters committed:
Merge pull request #729 from gaal/go

Go Style Guide

gaal committed:
Fix a logic error in an example.

PiperOrigin-RevId: 489255983

gaal committed:
Merge pull request #730 from gaal/go

Go: Fix a logic error in an example.

gaal committed:
Add Go CODEOWNERS.

gaal committed:
Link to the Go style guide from README.md.

gaal committed:
Internal Change.

PiperOrigin-RevId: 489430034

ujihisa committed:
Trivial typofix

s/Vimscript/Vim script/

gaal committed:
Internal Change.

PiperOrigin-RevId: 489956630

gaal committed:
Merge pull request #731 from gaal/go

Add link to the Go guide from README.md.

ujihisa committed:
Merge branch 'gh-pages' into patch-1

a-googler committed:
Project import generated by Copybara.

PiperOrigin-RevId: 493761232

Carrotman42 committed:
Internal Change.

PiperOrigin-RevId: 497970050

michael-projectx committed:
Adds C++20 headers so they aren't flagged as c-style headers

gpshead committed:
Vimscript -> Vim script : Merge pull request #734 from ujihisa/patch-1

Trivial typofix

matttproud committed:
Merge pull request #747 from Carrotman42/export_tmp

Internal Change.

tkruse committed:
NOBUG: update test dependencies to fix new build issues, github action fail

tkruse committed:
#209 Fix header order confusing warning due to ambiguous "complex.h"

"complex.h" was both a C99 System header and a old legacy C++ System header.
These days, it can be expected that the C header is meant,
so cpplint should treat it like that.

algonell committed:
fix: add explicit module names

pylint triggered:

pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.StandardError' ?) instead.
pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.Exception' ?) instead.
pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.BaseException' ?) instead.

Suggestion works out.

n3world committed:
Add support for multiple categories

Allow NOLINT and NOLINTNEXTLINE to take csv of categories to be ignored

gpshead committed:
Add explicit module names to pylintrc to support pylint >= 3.0

contributed by @algonell.

n3world committed:
lint: Remove unnecessary parens

tkruse committed:
Merge remote-tracking branch 'google/gh-pages' into develop

tkruse committed:
NOBUG: Use safer pull_request github trigger

aaronliu0130 committed:
Merge pull request #78 from groegeorg/master

jspricke committed:
Drop deprecated sre_compile usage (#214)

aaronliu0130 committed:
hopefully fix ci by bumping versions

aaronliu0130 committed:
dummy edit to trigger workflows

aaronliu0130 committed:
downgrade setuptools for now

aaronliu0130 committed:
disable auto-cancel

aaronliu0130 committed:
remove extraneous parens, continue failing fast and remove 3.12 which removed stuff

aaronliu0130 committed:
fix CI and only target 3.9 and pypy3.10

todo: modernize setup.py

Commit:
Fixed false positive when -readability/alt_tokens is used (#73)

Silenced a (probably illegitimate) pylint complain

Closes #73, which was the pull request I'm trying to merge here. PR-er
didn't provide proper authorship in their commit, so I'll try the GitHub
hidden email address below:

Co-authored-by: thibaudeau-irt <@users.noreply.github.com>

tirkarthi committed:
Refactor deprecated unittest aliases (#182)

For python 3.11 compatibility

aaronliu0130 committed:
Add new headers from C++23 and C23 (#239)

aaronliu0130 committed:
Modernize and bump version

Remove a bunch of readability-hampering extraneous support for the deprecated Python 2
Change flynt-able strings to f-strings
manually? convert stuff to f-string
modify pylintrc to make method-rgx not scream at __init__
replace deprecated assertEquals()s with assertEqual()s

aaronliu0130 committed:
hopefully make clitest windows-compatible

aaronliu0130 committed:
fix f-string spaces

aaronliu0130 committed:
fix "blank line contains whitespace" in test lol

aaronliu0130 committed:
confusion

aaronliu0130 committed:
Merge branch 'develop' into printing

aaronliu0130 committed:
update samples to match PR

aaronliu0130 committed:
remove transitive include exemption from IWYU

aaronliu0130 committed:
Add IWYU support for more stuff (#94 from geoffviola/printing)

aaronliu0130 committed:
Merge branch 'master' into mastery

aaronliu0130 committed:
Sync with dev branch

GrueMaster committed:
Fix recursive exclude Issue #114 (#115)

GrueMaster committed:
Fix recursive exclude Issue #114 (#115)

aaronliu0130 committed:
Revert "Fix recursive exclude Issue #114 (#115)"

already implemented, all it does is change a test file now

aaronliu0130 committed:
Revert "Fix recursive exclude Issue #114" (#242)

aaronliu0130 committed:
update language list

aaronliu0130 committed:
add back 3.11 target

jayvdb committed:
Add support for Python 3.12 (#243)

Removes 'lint' subcommand support from setup.py and replaces corresponding steps with the unmasked Python module commands

aaronliu0130 committed:
Merge from develop branch
